### PR TITLE
Fix countersignatory (to_remote) output redeemscript when anchors enabled

### PR DESCRIPTION
**Pull request by ksedgwic**
See https://github.com/lightning/bolts/blob/master/03-transactions.md#to_remote-output

### CODE REVIEW

**Comment by ksedgwic** [comment]
See https://github.com/lightning/bolts/blob/master/03-transactions.md#to_remote-output

File: lightning/src/ln/chan_utils.rs [hunk 1]
```diff
@@ -1130,7 +1141,11 @@ impl CommitmentTransaction {
 		let mut txouts: Vec<(TxOut, Option<&mut HTLCOutputInCommitment>)> = Vec::new();
 
 		if to_countersignatory_value_sat > 0 {
-			let script = script_for_p2wpkh(&countersignatory_pubkeys.payment_point);
+			let script = if opt_anchors {
+			    get_to_countersignatory_with_anchors_redeemscript(&countersignatory_pubkeys.payment_point).to_v0_p2wsh()
+			} else {
+			    script_for_p2wpkh(&countersignatory_pubkeys.payment_point)
```
Comment (devrandom) [inline lightning/src/ln/chan_utils.rs:1147]: maybe a good opportunity to rename this to match the `get_XXX_redeemscript` convention

**Review by devrandom — COMMENTED** [review]
let's update the unit test to cover this?

File: lightning/src/ln/chan_utils.rs (hunk 1 above)
Comment (ksedgwic) [reply lightning/src/ln/chan_utils.rs]: done

**Comment by ksedgwic** [comment]
> let's update the unit test to cover this?

done

File: lightning/src/ln/chan_utils.rs [hunk 2]
```diff
@@ -1489,7 +1505,8 @@ mod tests {
 			&mut htlcs_with_aux, &channel_parameters.as_holder_broadcastable()
 		);
 		assert_eq!(tx.built.transaction.output.len(), 4);
-
+		assert_eq!(tx.built.transaction.output[3].script_pubkey, get_to_countersignatory_with_anchors_redeemscript(&counterparty_pubkeys.payment_point).to_v0_p2wsh());
+
```
Comment (TheBlueMatt) [inline lightning/src/ln/chan_utils.rs:1509]: nit: trailing whitespace on this line.

**Review by TheBlueMatt — APPROVED** [review]

File: lightning/src/ln/chan_utils.rs (hunk 2 above)
Comment (ksedgwic) [reply lightning/src/ln/chan_utils.rs]: fixed

**Review by devrandom — APPROVED** [review]

**Pull request merged**